metrics.go

```diff
@@ -76,9 +76,6 @@ type Metrics struct {
 
 	// Number of times the Sampler failed to retrieve sampling strategy
 	SamplerQueryFailure metrics.Counter `metric:"sampler" tags:"state=failure,phase=query"`
-
-	// Number of times the Sampler failed to parse retrieved sampling strategy
-	SamplerParsingFailure metrics.Counter `metric:"sampler" tags:"state=failure,phase=parsing"`
```

Comment:
With the refactor, the parsing failure falls under UpdateFailure. Also, previously parsingfailure could occur when the strategy payload was correctly parsed but the strategy was invalid which should fall under UpdateFailure rather than ParseFailure.